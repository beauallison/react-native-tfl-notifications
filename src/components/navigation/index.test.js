import React from 'react';
import { shallow } from 'enzyme';
import Navigation, { LinesScreen, NotificationScreen } from './index';

describe('components/navigation', () => {
  it('should render navigation', () =>
    expect(shallow(<Navigation />)).toMatchSnapshot());

  it('should render the Lines screen', () =>
    expect(shallow(LinesScreen())).toMatchSnapshot());

  it('should render the Notifications screen', () =>
    expect(shallow(NotificationScreen())).toMatchSnapshot());
});
