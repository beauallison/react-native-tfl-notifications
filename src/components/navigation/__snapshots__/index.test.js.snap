// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/navigation should render navigation 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NavigationContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "navigation": Object {
        "dispatch": [Function],
        "goBack": [Function],
        "navigate": [Function],
        "setParams": [Function],
        "state": Object {
          "index": 0,
          "routes": Array [
            Object {
              "key": "Lines",
              "routeName": "Lines",
            },
            Object {
              "key": "Notifications",
              "routeName": "Notifications",
            },
          ],
        },
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "navigation": Object {
          "dispatch": [Function],
          "goBack": [Function],
          "navigate": [Function],
          "setParams": [Function],
          "state": Object {
            "index": 0,
            "routes": Array [
              Object {
                "key": "Lines",
                "routeName": "Lines",
              },
              Object {
                "key": "Notifications",
                "routeName": "Notifications",
              },
            ],
          },
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`components/navigation should render the Lines screen 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StationList
    data={undefined}
    onRefresh={
        Promise {
            "_40": 1,
            "_55": null,
            "_65": 0,
            "_72": Handler {
              "onFulfilled": [Function],
              "onRejected": [Function],
              "promise": Promise {
                "_40": 0,
                "_55": null,
                "_65": 0,
                "_72": null,
              },
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "data": undefined,
      "disableVirtualization": false,
      "horizontal": false,
      "initialNumToRender": 10,
      "keyExtractor": [Function],
      "maxToRenderPerBatch": 10,
      "numColumns": 1,
      "onEndReachedThreshold": 2,
      "onRefresh": [Function],
      "refreshing": true,
      "renderItem": [Function],
      "scrollEventThrottle": 50,
      "updateCellsBatchingPeriod": 50,
      "windowSize": 21,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "data": undefined,
        "disableVirtualization": false,
        "horizontal": false,
        "initialNumToRender": 10,
        "keyExtractor": [Function],
        "maxToRenderPerBatch": 10,
        "numColumns": 1,
        "onEndReachedThreshold": 2,
        "onRefresh": [Function],
        "refreshing": true,
        "renderItem": [Function],
        "scrollEventThrottle": 50,
        "updateCellsBatchingPeriod": 50,
        "windowSize": 21,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`components/navigation should render the Notifications screen 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationsList
    data={
        Array [
            Object {
              "ID": "BAKERLOO",
              "enabled": undefined,
              "station": "bakerloo",
              "updateStation": [Function],
            },
            Object {
              "ID": "CENTRAL",
              "enabled": undefined,
              "station": "central",
              "updateStation": [Function],
            },
            Object {
              "ID": "DISTRICT",
              "enabled": undefined,
              "station": "district",
              "updateStation": [Function],
            },
            Object {
              "ID": "CIRCLE",
              "enabled": undefined,
              "station": "circle",
              "updateStation": [Function],
            },
            Object {
              "ID": "HAMMERSMITH",
              "enabled": undefined,
              "station": "hammersmith-city",
              "updateStation": [Function],
            },
            Object {
              "ID": "JUBILEE",
              "enabled": undefined,
              "station": "jubilee",
              "updateStation": [Function],
            },
            Object {
              "ID": "LONDON",
              "enabled": undefined,
              "station": "london-overground",
              "updateStation": [Function],
            },
            Object {
              "ID": "METROPOLITAN",
              "enabled": undefined,
              "station": "metropolitan",
              "updateStation": [Function],
            },
            Object {
              "ID": "NORTHERN",
              "enabled": undefined,
              "station": "northern",
              "updateStation": [Function],
            },
            Object {
              "ID": "PICCADILLY",
              "enabled": undefined,
              "station": "piccadilly",
              "updateStation": [Function],
            },
            Object {
              "ID": "VICTORIA",
              "enabled": undefined,
              "station": "victoria",
              "updateStation": [Function],
            },
            Object {
              "ID": "WATERLOO",
              "enabled": undefined,
              "station": "waterloo-city",
              "updateStation": [Function],
            },
          ]
    }
    save={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <FlatList
          data={
                    Array [
                              Object {
                                "ID": "BAKERLOO",
                                "enabled": undefined,
                                "station": "bakerloo",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "CENTRAL",
                                "enabled": undefined,
                                "station": "central",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "DISTRICT",
                                "enabled": undefined,
                                "station": "district",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "CIRCLE",
                                "enabled": undefined,
                                "station": "circle",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "HAMMERSMITH",
                                "enabled": undefined,
                                "station": "hammersmith-city",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "JUBILEE",
                                "enabled": undefined,
                                "station": "jubilee",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "LONDON",
                                "enabled": undefined,
                                "station": "london-overground",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "METROPOLITAN",
                                "enabled": undefined,
                                "station": "metropolitan",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "NORTHERN",
                                "enabled": undefined,
                                "station": "northern",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "PICCADILLY",
                                "enabled": undefined,
                                "station": "piccadilly",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "VICTORIA",
                                "enabled": undefined,
                                "station": "victoria",
                                "updateStation": [Function],
                              },
                              Object {
                                "ID": "WATERLOO",
                                "enabled": undefined,
                                "station": "waterloo-city",
                                "updateStation": [Function],
                              },
                            ]
          }
          disableVirtualization={false}
          horizontal={false}
          initialNumToRender={10}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onEndReachedThreshold={2}
          renderItem={[Function]}
          scrollEventThrottle={50}
          updateCellsBatchingPeriod={50}
          windowSize={21}
/>,
        <SaveButton
          onPress={[Function]}
/>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [
            Object {
              "ID": "BAKERLOO",
              "enabled": undefined,
              "station": "bakerloo",
              "updateStation": [Function],
            },
            Object {
              "ID": "CENTRAL",
              "enabled": undefined,
              "station": "central",
              "updateStation": [Function],
            },
            Object {
              "ID": "DISTRICT",
              "enabled": undefined,
              "station": "district",
              "updateStation": [Function],
            },
            Object {
              "ID": "CIRCLE",
              "enabled": undefined,
              "station": "circle",
              "updateStation": [Function],
            },
            Object {
              "ID": "HAMMERSMITH",
              "enabled": undefined,
              "station": "hammersmith-city",
              "updateStation": [Function],
            },
            Object {
              "ID": "JUBILEE",
              "enabled": undefined,
              "station": "jubilee",
              "updateStation": [Function],
            },
            Object {
              "ID": "LONDON",
              "enabled": undefined,
              "station": "london-overground",
              "updateStation": [Function],
            },
            Object {
              "ID": "METROPOLITAN",
              "enabled": undefined,
              "station": "metropolitan",
              "updateStation": [Function],
            },
            Object {
              "ID": "NORTHERN",
              "enabled": undefined,
              "station": "northern",
              "updateStation": [Function],
            },
            Object {
              "ID": "PICCADILLY",
              "enabled": undefined,
              "station": "piccadilly",
              "updateStation": [Function],
            },
            Object {
              "ID": "VICTORIA",
              "enabled": undefined,
              "station": "victoria",
              "updateStation": [Function],
            },
            Object {
              "ID": "WATERLOO",
              "enabled": undefined,
              "station": "waterloo-city",
              "updateStation": [Function],
            },
          ],
          "disableVirtualization": false,
          "horizontal": false,
          "initialNumToRender": 10,
          "keyExtractor": [Function],
          "maxToRenderPerBatch": 10,
          "numColumns": 1,
          "onEndReachedThreshold": 2,
          "renderItem": [Function],
          "scrollEventThrottle": 50,
          "updateCellsBatchingPeriod": 50,
          "windowSize": 21,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onPress": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <FlatList
            data={
                        Array [
                                    Object {
                                      "ID": "BAKERLOO",
                                      "enabled": undefined,
                                      "station": "bakerloo",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "CENTRAL",
                                      "enabled": undefined,
                                      "station": "central",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "DISTRICT",
                                      "enabled": undefined,
                                      "station": "district",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "CIRCLE",
                                      "enabled": undefined,
                                      "station": "circle",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "HAMMERSMITH",
                                      "enabled": undefined,
                                      "station": "hammersmith-city",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "JUBILEE",
                                      "enabled": undefined,
                                      "station": "jubilee",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "LONDON",
                                      "enabled": undefined,
                                      "station": "london-overground",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "METROPOLITAN",
                                      "enabled": undefined,
                                      "station": "metropolitan",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "NORTHERN",
                                      "enabled": undefined,
                                      "station": "northern",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "PICCADILLY",
                                      "enabled": undefined,
                                      "station": "piccadilly",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "VICTORIA",
                                      "enabled": undefined,
                                      "station": "victoria",
                                      "updateStation": [Function],
                                    },
                                    Object {
                                      "ID": "WATERLOO",
                                      "enabled": undefined,
                                      "station": "waterloo-city",
                                      "updateStation": [Function],
                                    },
                                  ]
            }
            disableVirtualization={false}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
/>,
          <SaveButton
            onPress={[Function]}
/>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [
              Object {
                "ID": "BAKERLOO",
                "enabled": undefined,
                "station": "bakerloo",
                "updateStation": [Function],
              },
              Object {
                "ID": "CENTRAL",
                "enabled": undefined,
                "station": "central",
                "updateStation": [Function],
              },
              Object {
                "ID": "DISTRICT",
                "enabled": undefined,
                "station": "district",
                "updateStation": [Function],
              },
              Object {
                "ID": "CIRCLE",
                "enabled": undefined,
                "station": "circle",
                "updateStation": [Function],
              },
              Object {
                "ID": "HAMMERSMITH",
                "enabled": undefined,
                "station": "hammersmith-city",
                "updateStation": [Function],
              },
              Object {
                "ID": "JUBILEE",
                "enabled": undefined,
                "station": "jubilee",
                "updateStation": [Function],
              },
              Object {
                "ID": "LONDON",
                "enabled": undefined,
                "station": "london-overground",
                "updateStation": [Function],
              },
              Object {
                "ID": "METROPOLITAN",
                "enabled": undefined,
                "station": "metropolitan",
                "updateStation": [Function],
              },
              Object {
                "ID": "NORTHERN",
                "enabled": undefined,
                "station": "northern",
                "updateStation": [Function],
              },
              Object {
                "ID": "PICCADILLY",
                "enabled": undefined,
                "station": "piccadilly",
                "updateStation": [Function],
              },
              Object {
                "ID": "VICTORIA",
                "enabled": undefined,
                "station": "victoria",
                "updateStation": [Function],
              },
              Object {
                "ID": "WATERLOO",
                "enabled": undefined,
                "station": "waterloo-city",
                "updateStation": [Function],
              },
            ],
            "disableVirtualization": false,
            "horizontal": false,
            "initialNumToRender": 10,
            "keyExtractor": [Function],
            "maxToRenderPerBatch": 10,
            "numColumns": 1,
            "onEndReachedThreshold": 2,
            "renderItem": [Function],
            "scrollEventThrottle": 50,
            "updateCellsBatchingPeriod": 50,
            "windowSize": 21,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "onPress": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
